test/dynamo/test_autograd_function.py

```diff
@@ -467,32 +469,116 @@ def fn(a, b):
         self.assertEqual(res, MyMM.apply(a, a))
         self.assertEqual(cnt.frame_count, 1)
 
-    def test_graph_break_if_lifted_free_variable(self):
-        torch._dynamo.utils.counters.clear()
-        cnt = torch._dynamo.testing.CompileCounter()
-        delta = torch.randn(3)
+    def test_user_defined_object_as_input(self):
+        @dataclass
+        class Weird:
+            x: int
+            b: torch.Tensor
+            c: torch.Tensor
 
         class Foo(torch.autograd.Function):
             @staticmethod
-            def forward(ctx, x):
-                return x.clone(), (x + delta).clone()
+            def forward(ctx, x: torch.Tensor, weird: Weird, z: torch.Tensor):
+                ctx.save_for_backward(weird.b, weird.c)
+                return x.clone() * weird.b * weird.c
```

Comment:
Can you change this to `weird.b * weird.c * x.clone()`? (To test that the order things are accessed doesn't matter)